src/os/bluestore/BlueStore.cc

```diff
@@ -1621,10 +1621,15 @@ void BlueStore::SharedBlob::get_ref(uint64_t offset, uint32_t length)
 }
 
 void BlueStore::SharedBlob::put_ref(uint64_t offset, uint32_t length,
-  PExtentVector *r)
+				    PExtentVector *r,
+				    set<SharedBlob*> *maybe_unshared)
 {
   assert(persistent);
-  persistent->ref_map.put(offset, length, r);
+  bool maybe = false;
+  persistent->ref_map.put(offset, length, r, &maybe);
```

Comment:
you may want to pass maybe = nullptr if maybe_unshared == nullptr. This will speed up put() call a bit - no need to enumerate the whole ref_map if any.